src/SonarScanner.Shim/SonarScanner.Wrapper.cs

```diff
@@ -56,9 +56,16 @@ public class SonarScannerWrapper : ISonarScanner
 
         private const string SonarScannerVersion = "3.0.3.778";
 
+        private readonly ILogger logger;
+
+        public SonarScannerWrapper(ILogger logger)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
```

Comment:
One consequence of this form of null check is that we can't tell from the code coverage report whether we've forgotten to write a null check test (although those are fairly low value tests).

Reply:
@duncanp-sonar I think this is a problem linked to SonarQube. If you use tools which support branches like OpenCover (I think VS does) you'll see whether we covered the 2 branches of this line of code.